File: prob20.js
function factorial(n) { 
	if (factorial.values[n]) { 
		return factorial.values[n]; 
	} 
	return n * factorial(n - 1); 
}

factorial.values = {
	0: 1,
	1: 1
};

function sumDigits(toSum) { 
	var sum = 0; 
	while (toSum > 0) { 
		sum += toSum % 10;
		toSum = Math.floor(toSum / 10); 
	} 
	return sum; 
}

sumDigits(factorial(100));

File: prob20.py
FACT_MAP = {
	0: 1,
	1: 1
}

def factorial(n):
	if n in FACT_MAP:
		return FACT_MAP[n]
	return n * factorial(n - 1)

def sumDigits(toSum):
		value = 0
		while toSum > 0:
			value += toSum % 10
			toSum = toSum // 10
		return value

sumDigits(factorial(100))
